### ISSUE TITLE
unable to use _catch_errors for apply_prep function in pupept plan

### ISSUE THREAD

**Issue by srikanth2302** [puppetlabs/bolt#3290]
## Describe the Bug
apply_prep function unable to catch errors to go further in a plan. It says 'apply_prep' expects 1 argument, got 2"

## Expected Behavior
apply_prep function needs to catch error so that even if 1/10 nodes failed need to proceed further tasks in a plan.

## Steps to Reproduce
Tried below ways to catch errors in apply_prep function

Method 1:
$apply_prep_result = apply_prep($target_nodes,
{
  '_catch_errors' => true
})
Method 2:
$apply_prep_result = apply_prep($target_nodes,'_catch_errors' => true)

## Environment
 - Version - Puppet Enterprise 2021.7
 - Platform - Ubuntu 20.04

**Comment by donoghuc** [comment]
Bolt supports this as of https://github.com/puppetlabs/bolt/commit/9dcbd88a5661ffff42a60107c2f517d4249d0958 . I'm assuming you are not running this plan through bolt as a system package or gem, but rather through the orchestrator in PE?

**Comment by srikanth2302** [comment]
Yes.. this plan is run via orchestrator!! So will this not work in it? If not, can this be adjusted to handle via orchestrator as well?

**Comment by srikanth2302** [comment]
Can this be feasible to get ResultSet Info as well  Instead of BoltResultSet for apply_prep() to run via orchestrator PE Console Plan
